src/components/SignupForm/SignupForm.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import { Button } from '@material-ui/core';
+import SignUpFormDataInterface from './interfaces/SignupFormDataInterface';
+import ErrorMessageVisibleInterface from './interfaces/ErrorMessageVisibleInterface';
+import useSignupFormValidation from './useSignupFormValidation';
+import useFormFieldsConfig from './useFormFieldsConfig';
+import FormField from './FormField';
+import { auth } from '../../firebase';
+
+const SignupForm: React.FC = () => {
+  const initialFormData: SignUpFormDataInterface = {
+    name: '',
+    password: '',
+    repeatPassword: '',
+    email: ''
+  };
+
+  const initialErrorMessageVisible: ErrorMessageVisibleInterface = {
+    name: false,
+    email: false,
+    password: false,
+    repeatPassword: false
+  };
+
+  const [formData, setFormData] = useState(initialFormData);
+  const [errorMessageVisible, setErrorMessageVisible] = useState(initialErrorMessageVisible);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { error, setError, validateForm } = useSignupFormValidation(formData);
+  const formFieldsConfig = useFormFieldsConfig(formData, errorMessageVisible);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
+
+  const handleSubmit = async (e: React.ChangeEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (validateForm()) {
+      const { email, password, name } = formData;
+      try {
+        setIsLoading(true);
+        await auth.createUserWithEmailAndPassword(email, password);
+        await auth.currentUser?.updateProfile({
+          displayName: name
+        });
+      } catch (err) {
+        setError({
```

Comment:
const { email, password, name } = formData;
      try {
        setIsLoading(true);
        await auth.createUserWithEmailAndPassword(email, password);
        await auth.currentUser?.updateProfile({
          displayName: name
        });
      } catch (err) {
        setError({
          isError: true,
          errorText: err.message
        });
      }

it could be at least one or more separate function 
handleSubmit doing too much jobs - more than one